packages/datagateway-download/src/downloadConfirmation/downloadConfirmDialog.component.tsx

```diff
@@ -119,40 +136,74 @@ const DownloadConfirmDialog: React.FC<DownloadConfirmDialogProps> = (
   // Load the settings for use.
   const settings = React.useContext(DownloadSettingsContext);
 
-  // Set the default access method as the first access method
-  // defined in the configuration.
-  const defaultAccessMethod = Object.keys(settings.accessMethods)[0];
+  // Sorting and loading status.
+  const [statusMethods, setStatusMethods] = React.useState<
+    DownloadConfirmAccessMethod
+  >(
+    Object.keys(settings.accessMethods)
+      .map(key => {
+        return { method: key, methodInfo: settings.accessMethods[key] };
+      })
+      .reduce(
+        (obj, item) => ({
+          ...obj,
+          [item.method]: {
+            ...settings.accessMethods[item.method],
+
+            // Set disabled to be true by default.
+            disabled: true,
+            message: '',
+          },
+        }),
+        {}
+      )
```

Comment:
This code is a little unclear. We essentially want to take the `settings.accessMethods` object, and add the default `disabled` and `message` properties to each object right? I like functional programming and all that, but I think an easier to read and probably more efficient solution would just be a good ol' for...in loop

```
const defaultStatusMethods = {...settings.accessMethods};
for (const key in defaultStatusMethods) {
    defaultStatusMethods[key] = {...defaultStatusMethods[key], disabled: true, message: ''}
}
```

Reply:
@louise-davies I have simplified the code as you have shown in https://github.com/ral-facilities/datagateway/pull/141/commits/7facd345a991e683a4264bf7860b3f17406c16c3.